packages/did/src/DemoKeystore/DemoKeystore.ts

```diff
@@ -84,7 +103,8 @@ export class DemoKeystore
       )
     const keyType = DemoKeystore.getKeypairTypeForAlg(alg)
     try {
-      const keypair = this.signingKeyring.getPair(publicKey)
+      const encodedAddress = encodeSigningPublicKeyToAddress(publicKey, alg)
+      const keypair = this.signingKeyring.getPair(encodedAddress)
```

Comment:
Love your solution!
Although `getPairs().find(pair => u8aEq(pair.publicKey, publicKey)` could also have done the trick? (maybe additionally filtering for key type)
Alternatively you can set arbitrary metadata on a keyring pair, which if we put the alg and or public key here can help with filtering

Reply:
Yes, there could be alternative ways. This is also the way polkadotjs goes with, but instead of a function they have an object of closures to transform each public key to an address depending on the type. And this also provides constant time access to the right key, if present.